File: ArtistsViewController.m
//
//  ArtistsViewController.m
//  midwaybanking
//
//  Created by RONDINELE SANTOS DE MORAIS on 24/02/21.
//

#import "ArtistsViewController.h"

@interface ArtistsViewController()

@end

@implementation ArtistsViewController

- (void)viewWillAppear:(BOOL)animated {
  [super viewWillAppear:animated];
  [self.navigationController setNavigationBarHidden:true animated:false];
}

- (void)viewDidLoad {
  [super viewDidLoad];
  
  //  DEV
  //  NSURL *jsCodeLocation = [NSURL URLWithString:@"http://localhost:8081/index.bundle?platform=ios"];
  //
  //  RCTRootView *rootView = [[RCTRootView alloc] initWithBundleURL:jsCodeLocation
  //                                                      moduleName:@"RNMicroFrontArtist"
  //                                               initialProperties:nil
  //                                                   launchOptions:nil];
  //
  //  self.view = rootView;
  
  //  PROD
  NSBundle * bundle = [NSBundle bundleForClass:ArtistsViewController.class];
  NSString * srtPath = [bundle pathForResource: @"Bundle.bundle/RNMicroFrontArtist" ofType: @"jsbundle"];
  
  RCTRootView *rootView = [[RCTRootView alloc] initWithBundleURL:[NSURL URLWithString:srtPath]
                                                      moduleName:@"RNMicroFrontArtist"
                                               initialProperties:nil
                                                   launchOptions:nil];
  
  self.view = rootView;
}

@end

File: ArtistsViewController.swift
//
//  RNViewController.swift
//  MidwayOrc
//
//  Created by RONDINELE SANTOS DE MORAIS on 22/02/21.
//

import UIKit
import React

open class ArtistsViewController: UIViewController {
  
  open override func viewWillAppear(_ animated: Bool) {
    super.viewWillAppear(animated)
    navigationController?.setNavigationBarHidden(true, animated: false)
  }
  
  open override func viewDidLoad() {
    super.viewDidLoad()
    
//    let jsCodeLocation = URL(string: "http://localhost:8081/index.bundle?platform=ios")
//
//    let rootView = RCTRootView(
//      bundleURL: jsCodeLocation!,
//      moduleName: "RNMicroFrontArtist",
//      initialProperties: nil
//    )
//
//    self.view = rootView
    
    
    // Producao
    let bundle = Bundle(for: ArtistsViewController.self)

    let strPath = bundle.path(forResource: "Bundle.bundle/RNMicroFrontArtist", ofType: "jsbundle")!
    
    let rootView = RCTRootView(
      bundleURL: URL(string: strPath)!,
      moduleName: "RNMicroFrontArtist",
      initialProperties: nil
    )

    self.view = rootView
  }
}
